apps/trading/lib/hooks/use-games.ts

```diff
@@ -42,35 +92,75 @@ export const useGames = (teamId?: string, epochFrom?: number): GamesData => {
     skip: Boolean(epochFrom),
   });
 
-  let from = epochFrom;
-  if (!from && epochData) {
-    from = Number(epochData.epoch.id) - TEAMS_STATS_EPOCHS;
-    if (from < 1) from = 1; // make sure it's not negative
-  }
+  const [games, setGames] = useState<Game[] | undefined>(undefined);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<Error | ApolloError | undefined>(
+    undefined
+  );
 
-  const { data, loading, error } = useGamesQuery({
-    variables: {
-      epochFrom: from,
-      teamId: teamId,
-    },
-    skip: !from,
-    fetchPolicy: 'cache-and-network',
-    context: { isEnlargedTimeout: true },
-  });
+  const variables = useMemo(() => {
+    let from = epochFrom;
+    let to = epochTo;
+
+    if (epochData?.epoch.id && !epochFrom) {
+      const currentEpoch = Number(epochData.epoch.id);
+      from = currentEpoch - TEAMS_STATS_EPOCHS;
+      to = currentEpoch;
+    }
+
+    if (!from) return [];
+    return prepareVariables(teamId, from, to);
+  }, [epochData?.epoch.id, epochFrom, epochTo, teamId]);
+
+  useEffect(() => {
```

Comment:
@asiaznik can we get a comment here explaining why are dynamically creating a bunch of queries, rather than using the API in the expected way